app/controllers/serviceProviderListView.js

```diff
@@ -109,51 +118,65 @@ function parseResponse(){
 			//create table row with detail metadata
 			var row = Alloy.createController('serviceProviderRow', params).getView();	
 			
-			//TODO: re-work this as part of VCTMSVCS-123
 			//if provider has a crisis number, add it to crisis line table	
-			/*if(provider.crisisNumber){
+			if(provider.crisisNumber){
 				crisisHeaders.push(Alloy.createController('serviceProviderRow', {
 					orgName:provider.name,
 					crisis: provider.crisisNumber
 				}).getView());
-			}*/
-			
+			}
 			//iterate through all of the categories that service provider specializes in
 			_.each(provider.categories, function(category){
 				//iterate through list of table headers and add service provider to header if category matches
-				_.find(categoryDictionary, function(categoryDict){
-					if(categoryDict.id === category){
-						if(sections[categoryDict.id] == null){
-							sections[categoryDict.id] = Ti.UI.createTableViewSection({
-								title:categoryDict.id.id, 
-								headerView: Alloy.createController('TableViewHeader', {text:categoryDict.name}).getView()
-							});
-						}
-						sections[categoryDict.id].add(row);
-						return true;
-					}
-				});
-				//TODO: re-work this as part of VCTMSVCS-123
-				/*_.find(allHeaders, function(header){
-					if(header.title === category){
-						header.add(row);
-						return true;
-					}
-				});	*/
+				if(args.providerType === "general"){
+					addRowToDefinedSections(category, row);
+				}
+				else{
+					addRowToDynamicSections(sections, category, row);
+				}
+
 			});
 		});
 		var keys = Object.keys(sections);
-		var myHeaders = keys.map(function(v){return sections[v];});
+		var myHeaders = allHeaders.length > 0 ?  allHeaders : keys.map(function(v){return sections[v];});
 		originalMapAnnotations = $.map.annotations; //store original map points
 		//hide spinners
 		$.activityIndicator.hide();
 		//make tables visible
 		$.providerList.visible = true;
+		
 		//set table data
 		$.providerList.setData(myHeaders);
+		if(crisisMenu){
+			crisisMenu.setData(crisisHeaders);
+		}
 	}
 }
 
+function addRowToDynamicSections(sections, category, row){
```

Comment:
This function is pretty straight forward but I think it could still use comments.